### ISSUE TITLE
Preprocessor confused by multi-line quasiquotes with hashes at line start

### ISSUE THREAD

**Issue by thomasjm** [ndmitchell/hlint#844]
Sometimes I like to generate strings by using multi-line quasiquote expressions + interpolation like this:

```haskell
generatedStuff = [i|
Some static content:
#{generateStuff args}
|]
```

The fact that there's a `#` at the beginning of a line seems to confuse `hlint`, resulting in warnings like `Warning: unknown directive #{generateStuff`.

This seems related to #501, but it's a normal `.hs` file, not something unusual like literate Haskell. It seems like somewhere in the parsing process it's not understanding the quasiquote the way GHC does.

I confirmed that passing `--cpp-simple` makes the warnings go away, as that issue suggested.

**Comment by ndmitchell** [comment]
If you add `{-# LANGUAGE CPP #-}` to the top of that file, does GHC choke on it?

**Comment by thomasjm** [comment]
No, GHC still compiles it fine when I add `LANGUAGE CPP`.

**Comment by ndmitchell** [comment]
Thanks for the info. I'm curious how GHC copes! I'll investigate at some point in the future, probably after we've moved exclusively to GHC as a library (currently we have two parsing libraries), since the two different parse mechanisms will almost certainly complicate this bug.
